addons/headless/functions/fnc_modifyUnitsBlacklist.sqf

```diff
@@ -0,0 +1,50 @@
+#include "..\script_component.hpp"
+/*
+ * Author: johnb43
+ * Modifies which units are blacklisted from being transferred to HCs.
+ *
+ * Arguments:
+ * 0: Units <OBJECT, GROUP, ARRAY>
+ * 1: Add (true) or remove (false) from blacklist <BOOL> (default: true)
+ * 2: Transfer to server if blacklisted <BOOL> (default: true)
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [cursorObject, true] call ace_headless_fnc_modifyUnitsBlacklist
+ *
+ * Public: Yes
+ */
+
+params [["_units", objNull, [objNull, grpNull, []]], ["_blacklist", true, [false]], ["_transferToServer", true, [false]]];
+
+if !(_units isEqualType []) then {
+    _units = [_units];
+};
+
+_units = _units select {_x isEqualType objNull || {_x isEqualType grpNull}};
+_units = _units select {!isNull _x};
+
+if (_units isEqualTo []) exitWith {};
+
+private _groups = [];
+
+{
+    _x setVariable [QXGVAR(blacklist), _blacklist, true];
+
+    if (_blacklist && _transferToServer) then {
+        if (_x isEqualType objNull) then {
+            _groups pushBack _x;
+        } else {
+            _groups pushBack group _x
+        };
+    };
```

Comment:
@jonpas Do we want to apply the same filters here as in `FUNC(transferGroups)`? Under what conditions do we want to move the AI back to the server?

Reply:
I don't see why any filters would be needed here. As for condition when to move back, unsure, I don't see any other solution but an argument for that.

Reply:
I have found a condition: Only move units that were local to the HCs. I don't think any other filters or conditions are needed besides that.

Reply:
But I may want to keep them on HC and just prevent them from being transferred later.

Reply:
Which is why you have `_transferToServer` as an argument.